server/src/test/java/org/opensearch/snapshots/SnapshotUtilsTests.java

```diff
@@ -65,6 +65,8 @@ public void testIndexNameFiltering() {
         assertIndexNameFiltering(new String[] { "foo", "bar", "baz" }, new String[] { "-ba*" }, new String[] { "foo" });
         assertIndexNameFiltering(new String[] { "foo", "bar", "baz" }, new String[] { "+ba*" }, new String[] { "bar", "baz" });
         assertIndexNameFiltering(new String[] { "foo", "bar", "baz" }, new String[] { "+bar", "+foo" }, new String[] { "bar", "foo" });
+        assertIndexNameFiltering(new String[] { "foo", "bar", "baz" }, new String[] { "-bar", "b*" }, new String[] { "baz" });
+        assertIndexNameFiltering(new String[] { "foo", "bar", "baz" }, new String[] { "b*", "-bar" }, new String[] { "baz" });
```

Comment:
should we add a test here that tests both search indices to have a negation at the beginning and in that case, it should behave as normal and return the non-matching indices?

Reply:
I had thought that I saw one of the cases that looked like it might already test this case. I will check and if I do not see one add it in.

Reply:
Added